File: 125-valid-palindrome/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        d="".join(i for i in s.lower() if (96<ord(i) and ord(i)<=122) or (47<ord(i) and ord(i)<=57))
        return d[::]==d[::-1]

File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder st = new StringBuilder();
        for(char c: s.toCharArray()){
            if (Character.isLetterOrDigit(c)){
                st.append(Character.toLowerCase(c));
            }
        }
        s = st.toString();

        int left = 0;
        int right = s.length() -1;
        while(left<right){
            if(s.charAt(left) != s.charAt(right)){
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}